File: Задание 5/8.py
mc = 0 
cc = 0 
pn = None  
while True: 
    n = int(input("Введите натуральное число (0 для завершения): "))
    if n == 0: 
        break    
    if n == pn:  
        cc += 1
    else:
        mc = max(mc, cc)  
        cc = 1    
    pn = n
mc = max(mc, cc)
print(f"Наибольшее количество подряд идущих одинаковых элементов: {mc}")

File: 2 семестр /3 КТ/8.java
import java.util.Arrays;
public class VSUET {
    public static void main(String[] args) {
        String input = "56 65 74 100 99 68 86 180 90";
        String sorted = orderWeights(input);
        System.out.println(sorted);
    }
    public static String orderWeights(String str) {
        return Arrays.stream(str.trim().split("\\s+"))
                .sorted((a, b) -> {
                    int sumA = a.chars().map(c -> c - '0').sum();
                    int sumB = b.chars().map(c -> c - '0').sum();
                    if (sumA != sumB) {
                        return Integer.compare(sumA, sumB);
                    } else {
                        return a.compareTo(b);
                    }
                })
                .reduce((x, y) -> x + " " + y)
                .orElse("");
    }
}
